File: Animal Shelter Opdracht/C# versie/animal_shelter-abstract_interfaces-649f03d658d7546000a2cadc0e118ada4f610607/Animal Shelter/Animals/Animal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Animal_Shelter.Animals
{
    public class Animal
    {
        public string Name { get; private set; }
        public Gender Gender { get; private set; }
        public Reservor ReservedBy { get; private set; }

        public Animal(string name, Gender gender)
        {
            this.Name = name;
            this.Gender = gender;
        }

        public bool Reserve(string reservedBy)
        {
            if (this.ReservedBy == null)
            {
                this.ReservedBy = new Reservor(reservedBy, DateTime.Now);
                return true;
            }
            return false;
        }

        public override string ToString()
        {
            string reserved = "not reserved";
            if (this.ReservedBy != null)
            {
                reserved = $"reserved by {this.ReservedBy.Name}";
            }
            return $"{this.Name}, {this.Gender}, {reserved}";
        }
    }
}

File: src/Animals/Animal.java
package Animals;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Observable;

public class Animal
{
    public String Name;
    public Gender Gender;
    public Reservor ReservedBy;

    public int price;

    public Animal(String name, Animals.Gender gender, int price)
    {
        this.Name = name;
        this.Gender = gender;
        this.price = price;
    }

    public boolean Reserve(String reservedBy)
    {
        if (this.ReservedBy == null)
        {
            Date today = new Date();
            this.ReservedBy = new Reservor(reservedBy, today);
            return true;
        }
        return false;
    }

    public String ToString()
    {
        String reserved = "not reserved";
        if (this.ReservedBy != null)
        {
            reserved = "reserved by " + this.ReservedBy.Name;
        }
        return this.Name + " , " + this.Gender + " , " + reserved;
    }

    public String getName()
    {
        return Name;
    }

    public Animals.Gender getGender()
    {
        return Gender;
    }

    public Reservor getReservedBy()
    {
        return ReservedBy;
    }

    public void setName(String name)
    {
        Name = name;
    }

    public void setGender(Animals.Gender gender)
    {
        Gender = gender;
    }

    public void setReservedBy(Reservor reservedBy)
    {
        ReservedBy = reservedBy;
    }

}
